### DISCUSSION TITLE
What exactly does preprocessing_fn in smp.encoders do ?

### DISCUSSION THREAD

**Discussion by StepanVL** [qubvel/segmentation_models.pytorch#980] [Q&A]
I am training my model on image crops of size 384x384 pixels. 
Ideally, the full image should be passed through albumentations, including color normalization on a whole image. Then the `preprocessing_fn` should (or should not be applied). Then passed to training. 
For inference, full image is taken, color norm and resizing is done, `preprocessing_fn` should (or should not be applied) is passed to the model for inference.
Do I need to use `preprocessing_fn` during training or during inference? I want to aviod an inconcistency that happens that color is normalized (or something else happens) on a crop during training, but on the full image during inference. 

It seems that `preprocessing_fn` does the same as 

`color_norm = Compose([Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], )])`
`sample = color_norm(image=image)`
`color_norm_image = sample['image']`

**Comment by qubvel** [comment]
Hi @StepanVL! Thanks for the question! You can ignore `preprocessing_fn` for inference in case it was not used during training. It is usually just image normalization, but the same operation also can be done by any 3rd party library such as `albumentations`.
